pkg/build/reaper/reaper.go

```diff
@@ -30,78 +31,93 @@ type BuildConfigReaper struct {
 
 // Stop deletes the build configuration and all of the associated builds.
 func (reaper *BuildConfigReaper) Stop(namespace, name string, timeout time.Duration, gracePeriod *kapi.DeleteOptions) error {
-	noBcFound := false
-
-	// Add deletion pending annotation to the build config
-	err := unversioned.RetryOnConflict(unversioned.DefaultRetry, func() error {
-		bc, err := reaper.oc.BuildConfigs(namespace).Get(name)
-		if kerrors.IsNotFound(err) {
```

Comment:
I could not determine a reason for the special handling of not found. Any non-conflict error returned to RetryOnConflict appears to terminate retries immediately. The only effect of manually creating a NewNotFound seemed to be removing the "s" from the ultimate error message:
Error from server: buildconfig**s** "test" not found